### PR TITLE
Basic db int type support

### PR DESCRIPTION
**Pull request by guliashvili**
Added API and tests for int types in the database abstraction.

For now, it's just the  wrapper over the string type. In the following commits, will utilize specific database capabilites for additional performance.

### CODE REVIEW

**Comment by guliashvili** [comment]
ok to test

**Comment by fmanco** [comment]
ok to test

File: osquery/database/tests/database_tests.cpp [hunk 1]
```diff
@@ -23,31 +23,87 @@ namespace osquery {
...
+
+  EXPECT_TRUE(s.ok());
+  EXPECT_EQ(value, expected);
+}
+
+TEST_F(DatabaseTests, test_get_value_int) {
+  int expected = -1;
+  setDatabaseValue(kLogs, "int", expected);
+
+  int value;
```
Comment (akindyakov) [inline osquery/database/tests/database_tests.cpp]: To leave value unassigned is no a good idea, according to standard the value is unspecified.
You could use `auto value = int{};` or `int value{};` or `int value = 0;` which is the same.

File: osquery/database/tests/database_tests.cpp [hunk 2]
```diff
@@ -78,6 +134,24 @@ TEST_F(DatabaseTests, test_delete_values) {
   EXPECT_TRUE(value.empty());
 }
 
+TEST_F(DatabaseTests, test_delete_values_int) {
+  int expected = 0;
+  setDatabaseValue(kLogs, "k", 0);
```
Comment (akindyakov) [inline osquery/database/tests/database_tests.cpp]: 1. You probably mean `setDatabaseValue(kLogs, "k", expected);` here.
2. It  is better to use non-default  value here. Like 42 or whatever :)

File: osquery/database/tests/database_tests.cpp [hunk 3]
```diff
@@ -78,6 +134,24 @@ TEST_F(DatabaseTests, test_delete_values) {
...
+  auto s = deleteDatabaseValue(kLogs, "k");
+  EXPECT_TRUE(s.ok());
+
+  // Make sure the key has been deleted.
+  value = -5;
+  s = getDatabaseValue(kLogs, "k", value);
+  EXPECT_FALSE(s.ok());
+  EXPECT_EQ(value, -5);
+}
+
```
Comment (akindyakov) [inline osquery/database/tests/database_tests.cpp]: IMHO, it is good also to test writing to DB max acceptable integer numbers here, like `std::numeric_limits<int>::max()`. And to test writing to DB string values with symbols inside (`@!#$%?|/.,:\\^&*([])` and so on).

**Comment by guliashvili** [comment]
Thanks for review @akindyakov

**Comment by akindyakov** [comment]
Looks good to me

**Comment by mkareta** [comment]
Ship It
Alt text

**Pull request merged**